example/front_matter.dart

```diff
@@ -1,12 +1,12 @@
 import 'dart:io';
-import 'package:front_matter/front_matter.dart' as front_matter;
+import 'package:front_matter/front_matter.dart';
 
 // Example 1 - Parse a string.
 void example1() async {
   var file = new File('example/hello-world.md');
   var fileContents = await file.readAsString();
 
-  var doc = front_matter.parse(fileContents);
+  var doc = FrontMatterDocument.fromText(fileContents);
```

Comment:
Was this change really necessary? It introduces a breaking API change. Happy to discuss this in a separate issue (or PR) but I think this is the wrong PR for it.

Reply:
If you feel this PR isn't the place for it, I can revert it back to the way it is done in your version of the package.

Reply:
If you don't mind me asking, as I'm working on reverting it now, is there a reason you have the `parse` method pointing directly to the `parseContent` method, except for the defaultDelimiter bit?

Reply:
> is there a reason you have the `parse` method pointing directly to the `parseContent` method?

Just a functional programming style.

> Would you mind if I just put the contents of `parseContent` directly into the `parse` method?

I would, unfortunately. I've just pushed a new release `1.1.0` with some small changes. One of the changes is to rename the `parseContent` method to `parser`. I'd appreciate if you kept the current directory structure, with the two methods (`parse` and `parseFile`) exposed from `front_matter.dart` delegating out to the `parser()` from `parser.dart`.

I recommend rebasing with the latest commits from master and starting from there.

Reply:
While I'm considering how to implement my changes in line with the things you've mentioned, is the FrontMatterDocument constructor supposed to be public? Is it supposed to be used outside of the package itself?

Reply:
> Is it supposed to be used outside of the package itself?

At the moment, no. But this is subject to change.

Reply:
@MaterialTime the more I think about it, the less I'm convinced that this `FrontMatterDocument` to `String` method is possible. The method to generate the YAML string from a `Map` is outside the scope of this package, and better suited to the `yaml` package.

Reply:
I've looked around, and the yamlicious package might be a suitable alternative for generating the YAML portion of `value`. I hope that this can be used to resolve the issue of generating YAML strings being outside the scope of this package.